src/spot_skills/src/spot_skills/spot_arm_controller.py

```diff
@@ -25,14 +27,58 @@ def __init__(self, spot_manager: SpotManager):
         self._last_command_id = None  # ID of the most recent robot command to Spot
         self._last_trajectory = None  # Most recent trajectory sent to Spot
 
+        # Default: Send the maximum number of points at a time (250, per Spot SDK)
+        self.max_segment_len = 250
+
+    def send_trajectory_segment(self, segment: JointTrajectory) -> None:
+        """Command Spot to execute the given trajectory segment *all at once*.
+
+        :param      segment     Short trajectory over joint positions/velocities/times
+        """
+        assert len(segment) <= self.max_segment_len, "Segment too long!"
+
+        # Wait to send the segment until within 5 seconds of its start
+        wait_until_before_s = 5.0
+
+        segment_starts_in_s = segment.reference_timestamp.to_time_s() - time.time()
```

Comment:
I think this logic is wrong - Do we want relative to the *timestamp* or the *first point in the segment*?

Reply:
Addressed.